### PR TITLE
[29390] Introduce new work package change and commit

### PR DESCRIPTION
**Pull request by oliverguenther**
- [x] Introduces a basic `ChangeSet` that is HAL/Resource independent
- [x] Create a proxy wrapper around any resource to maintain access to both the pristine resource and a projected resource that behaves just like a saved resource (e.g., a HAL resource).
- [x] Removes the token based injections for wpCreate, wpEditing services

### CODE REVIEW

File: frontend/src/app/components/modals/editor/macro-wiki-include-page-modal/wiki-include-page-macro.modal.ts [hunk 1]
```diff
@@ -31,10 +31,6 @@ import {OpModalLocalsToken} from "core-components/op-modals/op-modal.service";
 import {AfterViewInit, ChangeDetectorRef, Component, ElementRef, Inject, ViewChild} from "@angular/core";
 import {OpModalLocalsMap} from "core-components/op-modals/op-modal.types";
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
-import {WorkPackageCreateService} from "core-components/wp-new/wp-create.service";
-import {IWorkPackageCreateServiceToken} from "core-components/wp-new/wp-create.service.interface";
-import {TypeResource} from "core-app/modules/hal/resources/type-resource";
-import {CurrentProjectService} from "core-components/projects/current-project.service";
```
Comment (openprojectci) [inline frontend/src/app/components/modals/editor/macro-wiki-include-page-modal/wiki-include-page-macro.modal.ts:34]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/modals/editor/macro-wiki-include-page-modal/wiki-include-page-macro.modal.ts#L34"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts [hunk 2]
```diff
@@ -2,7 +2,6 @@ import {Component, Input, OnDestroy, OnInit} from '@angular/core';
 import {componentDestroyed} from 'ng2-rx-componentdestroyed';
 import {takeUntil} from 'rxjs/operators';
 import {WorkPackageCacheService} from '../../work-packages/work-package-cache.service';
-import {combineLatest} from 'rxjs';
```
Comment (openprojectci) [inline frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts:5]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/work-packages/wp-relations-count/wp-watchers-count.component.ts#L5"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/wp-details/wp-details-toolbar.component.ts [hunk 3]
```diff
@@ -29,7 +29,6 @@ import {WorkPackageEditingService} from '../wp-edit-form/work-package-editing-se
 import {I18nService} from 'core-app/modules/common/i18n/i18n.service';
 import {Component, Inject, Input} from '@angular/core';
 import {WorkPackageResource} from 'core-app/modules/hal/resources/work-package-resource';
-import {IWorkPackageEditingServiceToken} from "core-components/wp-edit-form/work-package-editing.service.interface";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-details/wp-details-toolbar.component.ts:32]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/wp-details/wp-details-toolbar.component.ts#L32"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/wp-edit/wp-edit-field/wp-edit-field-group.directive.ts [hunk 4]
```diff
@@ -42,8 +42,6 @@ import {WorkPackageResource} from 'core-app/modules/hal/resources/work-package-r
 import {WorkPackageNotificationService} from '../wp-notification.service';
 import {WorkPackageCreateService} from './../../wp-new/wp-create.service';
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
-import {IWorkPackageEditingServiceToken} from "../../wp-edit-form/work-package-editing.service.interface";
-import {IWorkPackageCreateServiceToken} from "core-components/wp-new/wp-create.service.interface";
 import {WorkPackageViewSelectionService} from "core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-selection.service";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-edit/wp-edit-field/wp-edit-field-group.directive.ts:46]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/wp-edit/wp-edit-field/wp-edit-field-group.directive.ts#L46"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/components/wp-inline-create/wp-inline-create.component.ts [hunk 5]
```diff
@@ -56,15 +54,10 @@ import {IsolatedQuerySpace} from "core-app/modules/work_packages/query-space/iso
...
-import {IWorkPackageCreateServiceToken} from "core-components/wp-new/wp-create.service.interface";
-import {CurrentUserService} from "core-components/user/current-user.service";
 import {WorkPackageInlineCreateService} from "core-components/wp-inline-create/wp-inline-create.service";
 import {Subscription} from 'rxjs';
 import {WorkPackageViewColumnsService} from "core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-columns.service";
-import {
-  WorkPackageEvent,
-  WorkPackageEventsService
-} from "core-app/modules/work_packages/events/work-package-events.service";
+import {WorkPackageChangeset} from "core-components/wp-edit/work-package-changeset";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-inline-create/wp-inline-create.component.ts:61]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/wp-inline-create/wp-inline-create.component.ts#L61"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/editing-portal/edit-form-portal.component.ts [hunk 6]
```diff
@@ -17,9 +17,9 @@ import {
 } from "core-app/modules/fields/edit/edit-field.component";
 import {createLocalInjector} from "core-app/modules/fields/edit/editing-portal/edit-form-portal.injector";
 import {IFieldSchema} from "core-app/modules/fields/field.base";
-import {WorkPackageChangeset} from "core-components/wp-edit-form/work-package-changeset";
 import {EditFieldService, IEditFieldType} from "core-app/modules/fields/edit/edit-field.service";
-import {EditChangeset} from "core-app/modules/fields/changeset/edit-changeset";
+import {WorkPackageChangeset} from "core-components/wp-edit/work-package-changeset";
+import {ResourceChangeset} from "core-app/modules/fields/changeset/resource-changeset";
 import {HalResource} from "core-app/modules/hal/resources/hal-resource";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/editing-portal/edit-form-portal.component.ts:24]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/modules/fields/edit/editing-portal/edit-form-portal.component.ts#L24"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/modals/editor/macro-wp-button-modal/wp-button-macro.modal.ts#L38">frontend/src/app/components/modals/editor/macro-wp-button-modal/wp-button-macro.modal.ts#L38</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts#L38">frontend/src/app/components/wp-buttons/wp-status-button/wp-status-button.component.ts#L38</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/wp-custom-actions/wp-custom-actions/wp-custom-action.component.ts#L40">frontend/src/app/components/wp-custom-actions/wp-custom-actions/wp-custom-action.component.ts#L40</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts#L62">frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts#L62</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/modules/boards/board/board-list/board-list.component.ts#L55">frontend/src/app/modules/boards/board/board-list/board-list.component.ts#L55</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/1fff1312cfdf3a0cccabe6dc343ed5d7270cac9c/frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L37">frontend/src/app/modules/fields/edit/field-types/multi-select-edit-field.component.ts#L37</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by HDinger — APPROVED** [review]
Looks good, as far as I can tell. I did not have the time yet, to test it, but I trust on Travis to find major issues. Code wise it looks good. There are some minor code climate things, like unused imports or ToDos, but it is okay for me to let them in for now. 👍

**Pull request merged**